### ISSUE TITLE
Using Automatic Standardization in Image Generator Generates Much Worse Fit to Validation Data

### ISSUE THREAD

**Issue by penguin2121** [fchollet/keras#2622]
I'm using a 1.0 version of Keras I updated within the last week or so.  When I try to use the image data generator and set featurewise_center=True and featurewise_std_normalization=True to set the mean to 0 and the stdev to 1 of each image, the training goes fine, but the validation set either doesn't improve or only improves slightly.  If I turn thee off, the validation set converges at nearly the same rate as the train set.  How are these options implemented differently on the training/validation sets?  (the validation set I am using is defined in the fit_generator using validaitond_data=***).

Thanks!

**Comment by joelthchao** [comment]
Do you fit your validation set into another `ImageDataGenerator` as training set does and set `validation_data=val_datagen` in `fit_generator`?

**Comment by penguin2121** [comment]
I wasn't sure if the fit_generator forced the validation data through the same generator, but now that I'm thinking about it, it probably doesn't because that would mean random rotations/skews for it too...I'll try to implement a separate validation generator and see if that fixed it.

Thanks a lot!

**Comment by joelthchao** [comment]
@penguin2121 In `fit_generator`, val data won't pass through the generator for the train data, therefore you need to make another one for val data.

**Comment by penguin2121** [comment]
Do you also need to make a generator for the test data?  If the train/validation data is mean centered, will the test data be mean centered when you use predict?

**Comment by joelthchao** [comment]
Most of the time we will use generator which fits to train data to make sure train/test data has same distribution.

**Comment by penguin2121** [comment]
Great, thank you!
